harness/propertyHelper.js

```diff
@@ -1,13 +1,17 @@
 
 function isConfigurable(obj, name) {
+    var descriptor = Object.getOwnPropertyDescriptor(obj, name);
     try {
         delete obj[name];
     } catch (e) {
         if (!(e instanceof TypeError)) {
             $ERROR("Expected TypeError, got " + e);
         }
     }
-    return !Object.prototype.hasOwnProperty.call(obj, name);
+    var result = !Object.prototype.hasOwnProperty.call(obj, name);
+    // Restore property (no-op if not deleted)
+    Object.defineProperty(obj, name, descriptor);
```

Comment:
We've already discussed this before in other issues.

I don't think we should re-add the property. Once it's used, it should be consumed in the sense of atomic unit tests. To test anything in top of that, another set of object/properties would be recommended instead.

We can't avoid the side effect a configurable property will be deleted. This operation is not only restoring it, but triggering a `[[DefineOwnProperty]]` operation and this might cause other side effects in other test files. 

If I'm not wrong, @bterlson was in those other discussions about this.